### PR TITLE
feat: add JavaScript/Node.js based chatbot recipe

### PR DESCRIPTION
**Pull request by mhdawson**
I don't think we'd want to duplicate all of the examples but having a Node.js example for the chatbot and for the RAG recipe is what I'd think would be a good place to be.  This PR is the first step of adding the chatbot example in Node.js

It might need a few tweaks/testing of the framework stuff to build/package under quadlet but I have updated most things I think are needed and done what testing I could do without a bit more guidance.

### CODE REVIEW

**Comment by mhdawson** [comment]
@Gregory-Pereira as discussed FYI.

**Comment by mhdawson** [comment]
@Gregory-Pereira wondering what the next step might be on moving this forward?

File: recipes/natural_language_processing/chatbot-nodejs/ai-lab.yaml
Comment (Gregory-Pereira) [inline recipes/natural_language_processing/chatbot-nodejs/ai-lab.yaml]: Shouldnt this be:
```suggestion
        - llama-cpp
```
as per: https://github.com/containers/ai-lab-recipes/blob/main/recipes/natural_language_processing/chatbot/ai-lab.yaml#L12

File: recipes/natural_language_processing/chatbot-nodejs/provision/playbook.yml
Comment (Gregory-Pereira) [inline recipes/natural_language_processing/chatbot-nodejs/provision/playbook.yml:1]: You dont actually need this directory - it is for the E2E bootc framework which this will not need to make it into. You can remove this.

File: recipes/natural_language_processing/chatbot-nodejs/provision/requirements.yml
Comment (Gregory-Pereira) [inline recipes/natural_language_processing/chatbot-nodejs/provision/requirements.yml:1]: Same as above, remove this file + directory.

**Review by Gregory-Pereira — COMMENTED** [review]
Generally looks pretty close, just some small fixes. I will also want to test this locally before I approve for merging, unless you have done this already

File: recipes/natural_language_processing/chatbot-nodejs/provision/requirements.yml
Comment (mhdawson) [reply recipes/natural_language_processing/chatbot-nodejs/provision/requirements.yml:1]: @Gregory-Pereira removed the directory and the 2 files

File: recipes/natural_language_processing/chatbot-nodejs/ai-lab.yaml
Comment (mhdawson) [reply recipes/natural_language_processing/chatbot-nodejs/ai-lab.yaml]: Looks like it was changed since I put the PR together, accepted the suggestion.

**Comment by mhdawson** [comment]
@Gregory-Pereira I had tested it out to the extend that I could, but I think you testing locally would be good as a sanity check.  It was also quite a while ago that I ran that testing./

**Comment by rhatdan** [comment]
@Gregory-Pereira this is waiting on you?

**Review by Gregory-Pereira — APPROVED** [review]
/LGTM

**Pull request merged**
